src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java

```diff
@@ -263,6 +264,22 @@ public void testMultiFileFireErrors() throws Exception {
                 .isEqualTo(2);
     }
 
+    /**
+     * S2384 - Mutable members should not be stored or returned directly.
+     * Inspection is valid, a pure unit test is required as this condition can't be recreated in
+     * a test with checks and input file as none of the checks try to modify the fileExtensions.
+     */
+    @Test
+    public void testCopiedArrayIsReturned() {
+        final DummyFileSetCheck check = new DummyFileSetCheck();
+        final String[] extensions = {".tmp"};
+        check.setFileExtensions(extensions);
+        final String[] checkExtensions = TestUtil.getInternalState(check, "fileExtensions");
+        assertWithMessage("Extensions should be copied")
+                .that(checkExtensions)
+                .isNotSameInstanceAs(check.getFileExtensions());
+    }
```

Comment:
we can avoid reflection here. 
also, the setter supports varargs, so the code can be simplified
```java
@Test
public void testCopiedArrayIsReturned() {
    final DummyFileSetCheck check = new DummyFileSetCheck();
    check.setFileExtensions(".tmp");
    assertWithMessage("Extensions should be copied")
            .that(check.getFileExtensions())
            .isNotSameInstanceAs(check.getFileExtensions());
}
```

Reply:
Good suggestion, changes done.